### PR TITLE
Migrating configuration file to the new norm

### PR DESCRIPTION
**Pull request by l0k0ms**
### What does this PR do?

Migrates core-check configuration file in this repository to the new configuration file norm for more consistency: https://github.com/DataDog/integrations-core/blob/master/docs/dev/new_check_howto.md#configuration-file

### Motivation

After having migrated all integrations from integrations core / extras, this PR (hopefully) migrates the last configuration file.

### CODE REVIEW

**Review by ruthnaebeck — CHANGES_REQUESTED** [review]

**Review by ruthnaebeck — DISMISSED** [review]
👍 for docs

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 1]
```diff
@@ -1,77 +1,243 @@
...
+
 init_config:
 
+  ## @param docker_root - string - optional - default: /
+  ## Change the root directory for retrieving cgroup statistics. Useful when running inside a
+  ## container with host directories mounted on a different folder.
+  ## Example for the docker-dd-agent container:
+  ## docker_root: /host
+  #
+  # docker_root: /
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: this is legacy, should be removed, it's now configured at the agent level

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 2]
```diff
@@ -1,77 +1,243 @@
...
+
+  ## @param timeout - integer - optional - default: 5
+  ## Timeout in seconds for the connection to the docker daemon
+  #
+  # timeout: 5
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: this is legacy, should be removed, it's now configured at the agent level via `docker_query_timeout`

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 3]
```diff
@@ -1,77 +1,243 @@
...
+
+  ## @param api_version - string - optional - default: auto
+  ## The version of the API the client uses.
+  ## 'auto' picks the API version provided by the server.
+  #
+  # api_version: auto
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: ditto, legacy

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 4]
```diff
@@ -1,77 +1,243 @@
...
+
+  ## @param tls - boolean - optional - default: false
+  ## Set to true to enable TLS encryption while communicating
+  ## with the Docker API.
+  #
+  # tls: false
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: ditto

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 5]
```diff
@@ -1,77 +1,243 @@
...
+
+  ## @param tls_client_cert - string - optional
+  ## Path to the client certificate file (client-cert.pem).
+  #
+  # tls_client_cert: <CLIENT_CERT_PATH>
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: ditto

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 6]
```diff
@@ -1,77 +1,243 @@
...
+
+  ## @param tls_client_key - string - optional
+  ## Path to the TLS client key file (client-key.pem).
+  #
+  # tls_client_key: <CLIENT_KEY_PATH>
+
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: ditto

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 7]
```diff
@@ -1,77 +1,243 @@
...
+  ## @param tls_cacert - string - optional
+  ## Path to the TLS CA certificate file (ca.pem).
+  #
+  # tls_cacert: <TLS_CA_CERT_PATH>
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: ditto

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 8]
```diff
@@ -1,77 +1,243 @@
...
+
+  ## @param tls_verify - boolean - optional - default: true
+  ## Set this variable to false to disable TLS verification.
+  #
+  # tls_verify: true
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: ditto

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 9]
```diff
@@ -1,77 +1,243 @@
...
+
+  ## @param init_retry_interval - integer - optional - default: 0
+  ## Initialization retries if the Agent is expected to start before Docker,
+  ## Use these settings to configure the retry policy.
+  ## init_retry_interval defines how long (in seconds) the Docker client
+  ## waits before retrying initialization.
+  #
+  # init_retry_interval: 0
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: ditto

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 10]
```diff
@@ -1,77 +1,243 @@
...
+
+  ## @param init_retries - integer - optional - default: 0
+  ## init_retries configures how many retries are made before failing permanently.
+  #
+  # init_retries: 0
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: ditto

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 11]
```diff
@@ -1,77 +1,243 @@
...
 instances:
-  - ## The agent honors the DOCKER_HOST, DOCKER_CERT_PATH and DOCKER_TLS_VERIFY
-    ## environment variables to setup the connection to the server.
-    ## See https://docs.docker.com/engine/reference/commandline/cli/#environment-variables
 
+    ## @param url - string - required
+    ## URL of the Docker daemon socket to reach the Docker API. HTTP/HTTPS also works.
+    ## Warning: if that's a non-local daemon, Datadog cannot collect performance metrics.
+    #
+  - url: "unix://var/run/docker.sock"
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: ditto, agent level

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 12]
```diff
@@ -1,77 +1,243 @@
...
 
-    # Collect images stats
-    # Number of available active images and intermediate images as gauges.
-    # Defaults to false.
+    ## @param custom_cgroups - boolean - optional - default: false
+    ## Set to true if you use a custom cgroup hierarchy for this instance.
+    ## Enabling this option modifies how the Agent inspects containers and causes
+    ## some overhead - if you run a high volume of containers the Agent may timeout.
+    #
+    # custom_cgroups: false
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: legacy

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 13]
```diff
@@ -1,77 +1,243 @@
...
+    ## Container health checks are available starting with Docker 1.12. Enabling it with older
+    ## versions results in an UNKNOWN state for the service check.
+    ##
+    ## You must white-list the containers you wish to submit health service checks for.
+    ## To white-list containers use the <TAG_NAME>:<VALUE> format.
+    ## For the list of supported tags, see the performance_tags option.
+    ## Example: "docker_image:tomcat" or "container_name:web_front_nginx"
+    #
+    # health_service_check_whitelist:
+    #   - <TAG_NAME>:<VALUE>
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: legacy

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 14]
```diff
@@ -1,77 +1,243 @@
...
+
+    ## @param collect_container_count - boolean - optional - default: false
+    ## Set to true to collect the container count tagged by state (running, paused, exited, dead).
     #
-    # collect_images_stats: true
+    # collect_container_count: false
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: legacy

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 15]
```diff
@@ -1,77 +1,243 @@
...
+    ## To exclude containers use the <TAG_NAME>:<VALUE> format.
+    ##
+    ## Example - exclude all, except Ubuntu and Debian:
+    ## exclude: ["docker_image:.*"]
+    ## include: ["docker_image:ubuntu", "docker_image:debian"]
+    ##
+    ## By default, pause containers are excluded based on their image (Fargate, Kubernetes, OpenShift, etc.)
+    #
+    # exclude:
+    #   - <TYPE>:<VALUE>
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: legacy, now at agent level

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 16]
```diff
@@ -1,77 +1,243 @@
...
+
+    ## @param include - list of key:value element - optional
+    ## Include containers based on their tags. The parameter value is a regex on the tags.
+    ## See the exclude option above for an example.
+    #
+    # include:
+    #   - <TAG_NAME>:<VALUE>
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: legacy, now at agent level

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 17]
```diff
@@ -1,77 +1,243 @@
...
+    #   - <KEY_1>:<VALUE_1>
+    #   - <KEY_2>:<VALUE_2>
 
-    # You can add extra tags to your Docker metrics with the tags list option.
-    # Example: ["extra_tag", "env:testing"]
+    ## @param ecs_tags - boolean - optional - default: true
+    ## If the Agent is running in an Amazon ECS task, the container metrics are tagged with the ECS task name and version.
     #
-    # tags: []
+    # ecs_tags: true
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: legacy

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 18]
```diff
@@ -1,77 +1,243 @@
...
+    ##
+    ## Available tags for performance metrics:
+    ##   - image_name: Name of the image (example: "nginx")
+    ##   - image_tag: Tag of the image (example: "latest")
+    ##   - docker_image: LEGACY. The full image name:tag string (example: "nginx:latest")
+    ##   - container_name: Name of the container (example: "boring_euclid")
+    ##   - container_command: Command ran by the container (example: "echo 1")
+    ##   - container_id: Id of the container
+    #
+    # performance_tags: ["container_name", "image_name", "image_tag", "docker_image"]
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: legacy

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 19]
```diff
@@ -1,77 +1,243 @@
...
+
+    ## @param container_tags - list of strings - optional - default: ["image_name", "image_tag", "docker_image"]
+    ## Tags for container count metrics.
+    ## Available:
+    ##   - image_name: Name of the image (example: "nginx")
+    ##   - image_tag: Tag of the image (example: "latest")
+    ##   - docker_image: LEGACY. The full image name:tag string (example: "nginx:latest")
+    ##   - container_command: Command ran by the container (example: "echo 1")
+    #
+    # container_tags: ["image_name", "image_tag", "docker_image"]
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: legacy

File: cmd/agent/dist/conf.d/docker.d/conf.yaml.example [hunk 20]
```diff
@@ -1,77 +1,243 @@
...
+
+    ## @param event_attributes_as_tags - list of strings - optional - default: none
+    ## Option to tag Docker metrics with container label names listed.
+    ## Takes precedence over docker_labels_as_tags for Docker metrics.
+    ## Only use if you want different labels tagged between Autodiscovery and Docker daemon metrics.
+    ## Example:
+    ##  event_attributes_as_tags: ["signal"]
+    #
+    # event_attributes_as_tags: ["LABEL_NAME"]
```
Comment (mfpierre) [inline cmd/agent/dist/conf.d/docker.d/conf.yaml.example]: legacy

**Review by mfpierre — CHANGES_REQUESTED** [review]
Docker being a corecheck now and rewritten from scratch a lot of config parameter are no longer relevant or at the agent level directly, for reference: https://github.com/DataDog/datadog-agent/blob/f5e1876f04ef1acb5de5ee90cf82b60dd88e454a/pkg/collector/corechecks/containers/docker.go#L39-L51

**Review by mfpierre — DISMISSED** [review]
LGTM for container configs 🙇

File: cmd/agent/dist/conf.d/disk.d/conf.yaml.default [hunk 21]
```diff
@@ -1,53 +1,162 @@
...
+    ## Add rootfs here to exclude.
+    #
     # excluded_filesystems:
     #   - tmpfs
 
-    # The (optional) excluded_disks parameter will instruct the check to
-    # ignore this list of disks.
-    # Linux example:
+    ## @param excluded_disks - list of strings - optional
+    ##  The `excluded_disks` parameter instructs the check to
```
Comment (FlorianVeaux) [inline cmd/agent/dist/conf.d/disk.d/conf.yaml.default]:
```suggestion
    ## The `excluded_disks` parameter instructs the check to
```

File: cmd/agent/dist/conf.d/disk.d/conf.yaml.default [hunk 22]
```diff
@@ -1,53 +1,162 @@
...
-    # The (optional) tag_by_filesystem parameter will instruct the check to
-    # tag all disks with their filesystem (for ex: filesystem:nfs).
-    # tag_by_filesystem: false
-    #
-    # The (optional) excluded_mountpoint_re parameter will instruct the check to
-    # ignore all mountpoints matching this regex.
-    # excluded_mountpoint_re: /mnt/somebody-elses-problem.*
+    # excluded_mountpoint_re: <MOUNT_POINT_REGEX>
+
+    ## @param all_partitions - boolean - optional
```
Comment (FlorianVeaux) [inline cmd/agent/dist/conf.d/disk.d/conf.yaml.default]:
```suggestion
    ## @param all_partitions - boolean - optional - default: false
```

File: cmd/agent/dist/conf.d/disk.d/conf.yaml.default [hunk 23]
```diff
@@ -1,53 +1,162 @@
...
+    ## Instruct the check to collect from partitions even without device names.
+    ## Setting `use_mount` to true is strongly recommended in this case.
     #
-    # The (optional) all_partitions parameter will instruct the check to
-    # get metrics for all partitions. use_mount should be set to yes (to avoid
-    # collecting empty device names) when using this option.
     # all_partitions: false
+
+    ## @param tags - list of key:value string - optional
```
Comment (FlorianVeaux) [inline cmd/agent/dist/conf.d/disk.d/conf.yaml.default]:
```suggestion
    ## @param tags - list of key:value strings - optional
```

File: cmd/agent/dist/conf.d/disk.d/conf.yaml.default [hunk 24]
```diff
@@ -1,53 +1,162 @@
...
+    ##
+    ## read-write -> OK
+    ## read-only  -> CRITICAL
+    ## other      -> UNKNOWN
     #
-    # The (optional) service_check_rw parameter notifies when one of the partitions is in
-    # read-only mode
     # service_check_rw: false
+
+    ## @param tag_by_filesystem - boolean - optional
```
Comment (FlorianVeaux) [inline cmd/agent/dist/conf.d/disk.d/conf.yaml.default]:
```suggestion
    ## @param tag_by_filesystem - boolean - optional - default: false
```

File: cmd/agent/dist/conf.d/jmx.d/conf.yaml.example [hunk 25]
```diff
@@ -1,72 +1,148 @@
...
-  #           - excluded_bean
-  #     - include:
-  #         domain_regex: regex_on_domain
-  #       exclude:
-  #         bean_regex:
-  #           - regex_on_excluded_bean
+
+  -
+
+    ## @param host - string - optional
```
Comment (FlorianVeaux) [inline cmd/agent/dist/conf.d/jmx.d/conf.yaml.example]:
```suggestion
    ## @param host - string - optional - default: localhost
```

File: cmd/agent/dist/conf.d/disk.d/conf.yaml.default (hunk 23 above)
Comment (FlorianVeaux) [reply cmd/agent/dist/conf.d/disk.d/conf.yaml.default]: The rest of the PR uses `list of key:value elements`, which one should be prefered?

File: cmd/agent/dist/conf.d/jmx.d/conf.yaml.example [hunk 26]
```diff
@@ -1,72 +1,148 @@
...
+    ## Cassandra host to connect to
```
Comment (FlorianVeaux) [inline cmd/agent/dist/conf.d/jmx.d/conf.yaml.example]: Cassandra?

File: cmd/agent/dist/conf.d/jmx.d/conf.yaml.example [hunk 27]
```diff
@@ -1,72 +1,148 @@
...
+    #
+    # host: localhost
+
+    ## @param port - integer - required
+    ## Cassandra port to connect to
```
Comment (FlorianVeaux) [inline cmd/agent/dist/conf.d/jmx.d/conf.yaml.example]: Cassandra?

File: cmd/agent/dist/conf.d/network.d/conf.yaml.default [hunk 28]
```diff
@@ -1,33 +1,91 @@
...
 
+    ## @param conntrack_path - string - optional
+    ## Linux only.
+    ## The location of the conntrack executable for collecting stats from conntrack -S.
+    ## This is run with sudo, so an entry needs to be added to the sudoers file.
+    ## By default, these metrics are not sent.
+    #
+    # conntrack_path: /usr/sbin/conntrack
+
+    ## @param whitelist_conntrack_metrics - list of string - optional - default: ["max", "count"]
```
Comment (FlorianVeaux) [inline cmd/agent/dist/conf.d/network.d/conf.yaml.default]:
```suggestion
    ## @param whitelist_conntrack_metrics - list of strings - optional - default: ["max", "count"]
```

**Review by FlorianVeaux — CHANGES_REQUESTED** [review]

File: cmd/agent/dist/conf.d/jmx.d/conf.yaml.example (hunk 26 above)
Comment (l0k0ms) [reply cmd/agent/dist/conf.d/jmx.d/conf.yaml.example]:
```suggestion
    ## JMX host to connect to
```

File: cmd/agent/dist/conf.d/jmx.d/conf.yaml.example (hunk 27 above)
Comment (l0k0ms) [inline cmd/agent/dist/conf.d/jmx.d/conf.yaml.example]:
```suggestion
    ## JMX port to connect to.
```

**Review by FlorianVeaux — DISMISSED** [review]

File: cmd/agent/dist/conf.d/disk.d/conf.yaml.default [hunk 29]
```diff
@@ -1,53 +1,162 @@
...
-    #  "c:": volume:boot
-    #
-    # The (optional) tags parameter allows you to customize tags for the instance
     # tags:
-    #   - optional_tag1
-    #   - optional_tag2
+    #   - <KEY_1>:<VALUE_1>
+    #   - <KEY_2>:<VALUE_2>
+
+    ## @param file_system_whitelist - list of regex - optional
```
Comment (hush-hush) [inline cmd/agent/dist/conf.d/disk.d/conf.yaml.default]: This parameter is only available in the Python version of the disk check. The Go version is more limited.
The Python version has it's own default configuration https://github.com/DataDog/integrations-core/blob/master/disk/datadog_checks/disk/data/conf.yaml.default.

The default configuration from integration is shipped by default. I think this file should only reflect the option available in the Go version of the check (with a disclaimer at the beginning maybe ?).

File: cmd/agent/dist/conf.d/network.d/conf.yaml.default [hunk 30]
```diff
@@ -1,33 +1,91 @@
...
+    #
     # collect_count_metrics: false
 
+    ## @param conntrack_path - string - optional
+    ## Linux only.
+    ## The location of the conntrack executable for collecting stats from conntrack -S.
+    ## This is run with sudo, so an entry needs to be added to the sudoers file.
+    ## By default, these metrics are not sent.
+    #
+    # conntrack_path: /usr/sbin/conntrack
```
Comment (hush-hush) [inline cmd/agent/dist/conf.d/network.d/conf.yaml.default]: same here than for the disk check: some option are only available in the Python version of the check.

File: cmd/agent/dist/conf.d/ntp.d/conf.yaml.default [hunk 31]
```diff
@@ -1,19 +1,45 @@
...
-    #  - 3.europe.pool.ntp.org
-    # port: 123
     # version: 3
+
+    ## @param timeout - integer - optional - default: 5
+    ## The timeout for connecting to the NTP server in second.
+    #
     # timeout: 5
+
+    ## @param tags - list of key:value elements - optional
```
Comment (hush-hush) [inline cmd/agent/dist/conf.d/ntp.d/conf.yaml.default]: I don't think that NTP check Go version supports tags. I think we should support it (created a card) but in the meantime we should remove this section.

**Review by hush-hush — CHANGES_REQUESTED** [review]

File: cmd/agent/dist/conf.d/disk.d/conf.yaml.default (hunk 29 above)
Comment (hush-hush) [reply cmd/agent/dist/conf.d/disk.d/conf.yaml.default]: Many option here are still only available in the python version:

- `file_system_blacklist`
- `device_whitelist`
- `mount_point_whitelist`
...

File: cmd/agent/dist/conf.d/network.d/conf.yaml.default (hunk 30 above)
Comment (hush-hush) [reply cmd/agent/dist/conf.d/network.d/conf.yaml.default]: Same here

**Comment by olivielpeau** [comment]
Moving to 6.15.0 milestone as 6.14.0 freeze is happening soon, @l0k0ms if this is ready to merge earlier let us know

File: cmd/agent/dist/conf.d/systemd.d/conf.yaml.example [hunk 32]
```diff
@@ -8,18 +8,29 @@ instances:
   - unit_names:
       - <UNIT_NAME>
 
+    ## @param unit_regexes - list of strings - optional
```
Comment (hush-hush) [inline cmd/agent/dist/conf.d/systemd.d/conf.yaml.example]: I can't find this being used anywhere. Where does this comes from ?

Comment (l0k0ms) [reply cmd/agent/dist/conf.d/systemd.d/conf.yaml.example]: you right, it's a miss.

**Review by hush-hush — APPROVED** [review]

**Pull request merged**